src/DataStructures/LeviCivitaIterator.hpp

```diff
@@ -61,21 +62,78 @@ class LeviCivitaIterator {
   }
 
   // Return the current (multi-index) Index, an ordered list of indices
-  const Index<Dim>& operator()() const noexcept {
+  const cpp17::array<size_t, Dim>& operator()() const noexcept {
     return indexes_[permutation_];
   }
 
   // Return a specific index from the current (multi-index) Index
-  const size_t& operator[](size_t i) {
-    return indexes_[permutation_][i];
-  }
+  const size_t& operator[](size_t i) { return indexes_[permutation_][i]; }
 
   // Return the sign of the Levi-Civita symbol with these indices
-  int sign() const noexcept { return signs_[permutation_]; }
+  int sign() const noexcept { return signs_[permutation_]; };
+
+  // Initialize the indexes
+  static constexpr cpp17::array<cpp17::array<size_t, Dim>, factorial(Dim)>
+  indexes() noexcept {
+    cpp17::array<cpp17::array<size_t, Dim>, factorial(Dim)> indexes{};
+    cpp17::array<size_t, Dim> index{};
+
+    // Add the initial index permutation to the list
+    // Implement this myself, since iota is not constexpr
```

Comment:
Just implement a constexpr iota like we discussed...

Reply:
Thanks, @nilsdeppe. I'm working on this. Turns out, I need to reimplement the following functions as constexpr: std::iota, std::accumulate, std::swap, std::iter_swap, std::reverse, std::next_permutation, none of which are constexpr. Currently debugging my reimplementation of std::next_permutation.

Is there a standard place to look for the implementations of these functions? I looked at cppreference.com for them.

Reply:
Just want to ask: are you all sure computing these Levi-Civita symbols at compile time is worth it? I could have a runtime version ready to go in no time. Not obvious to me that it's going to make any noticeable difference in performance; we're talking about computing 24 ints.

Reply:
Hey @geoffrey4444! You can look at libc++ [here](https://github.com/llvm-mirror/libcxx/tree/master/include) or libstdc++ [here](https://github.com/gcc-mirror/gcc/tree/master/libstdc%2B%2B-v3/include/std).

Probably not a huge difference for this case, but we will generally want these functions in a constexpr context in the future :)

Reply:
Thanks again, @nilsdeppe! I used libc++, because it has the same open-source license as we do.